app/src/main/java/com/example/cheatchess/MainActivity.kt

```diff
@@ -61,21 +71,10 @@ class MainActivity : AppCompatActivity(), ChessDelegate {
                 return@launch
             }
             if (response.isSuccessful && response.body() != null)
-                Log.d("MainActivity", response.body().toString())
+                binding.tvEval.text = response.body()!!.data
```

Comment:
I don't usually like the !! as it will crash if response.body is null. Can it change after you check it?

Reply:
I don't think it can change, because I am declaring a new response variable each time I call the API, if there is a new unsuccessful response it will be assigned to a new var and will be checked.

Reply:
There is a better way to express this. Do you know how to use let?

Reply:
Yes, so here should I do:
 if (response.isSuccessful) response.body()?.let { binding.tvEval.txt = it.data }

Reply:
Crashing isn't always a bad thing. What is the proper behavior of the app, in your mind?

Reply:
I want the user to know that if it's an internet issue, or a server issue, but that he is still able to set up the board offline